### PR TITLE
feat: enable advanced threat protection

### LINKED ISSUE #17: feat: enable advanced threat protection

### PR DESCRIPTION
**Pull request by sebastianlolv**
Closes https://github.com/equinor/terraform-azurerm-sql/issues/17

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -36,6 +36,11 @@ resource "azurerm_storage_account" "this" {
   }
 }
 
+resource "azurerm_advanced_threat_protection" "this" {
+  target_resource_id = azurerm_storage_account.this.id
```
Comment (hknutsen) [inline main.tf]: Should be enabled for the SQL server 🙂 

```suggestion
  target_resource_id = azurerm_mssql_server.this.id
```

File: variables.tf [hunk 2]
```diff
@@ -50,3 +50,9 @@ variable "firewall_rules" {
   description = "A map of IP address ranges that should be able to access this SQL Server."
   type        = map(tuple([string, string]))
 }
+
+variable "threat_protection_enabled" {
+  description = "Is threat protection (Microsoft Defender for Storage) enabled for this Storage Account?"
```
Comment (hknutsen) [inline variables.tf]:
```suggestion
  description = "Is threat protection (Microsoft Defender for SQL) enabled for this SQL Server?"
```

**Review by hknutsen — CHANGES_REQUESTED** [review]

**Comment by hknutsen** [comment]
@sebastianlolv Could you also merge the latest commits from branch `main` into this branch? I see the docs workflow is failing, which I believe is a result of this branch missing the latest commits.

**Comment by hknutsen** [comment]
When I run the tests locally, I get the following error:

```
╷
│ Error: checking for presence of existing Advanced Threat Protection for "/subscriptions/ce8c6461-a20d-47a3-884c-6659cb74e53f/resourceGroups/rg-16070686d2718ebd-test/providers/Microsoft.Sql/servers/sql-16070686d2718ebd-test": security.AdvancedThreatProtectionClient#Get: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="Not Supported" Message="Azure Defender for Data Services settings are not supported on resources of type servers"
│ 
│   with module.sql.azurerm_advanced_threat_protection.this,
│   on ../../main.tf line 39, in resource "azurerm_advanced_threat_protection" "this":
│   39: resource "azurerm_advanced_threat_protection" "this" {
│ 
╵
```

When I navigate to the created SQL server in the Azure Portal, I can see that Microsoft Defender for SQL is enabled, which includes Vulnerability Assessment and Advanced Data Protection:

bilde

When I disable Microsoft Defender for SQL in the Azure Portal, all settings related to Vulnerability Assessment and Advanced Data Protection are greyed out:

bilde


It seems like when we enabled Vulnerability Assessment using Terraform, it also enables Advanced Data Protection, since these two features seem to go hand in hand for Azure SQL servers.

@sebastianlolv If you agree with my observation, please close this PR 🙂

**Comment by sebastianlolv** [comment]
Looks good 👍 Closing.

**Pull request closed without merging**